File: test/premake5.ts
import { workspace } from "../src/generator/index.ts";

export default workspace("MyWorkspace", (w) => {
	w.configurations("Debug", "Release");

	w.project("MyProject", (p) => {
		p.kind("ConsoleApp")
		p.language("C++")
		p.characterSet("MBCS")
		p.cppDialect("C++23")
		p.targetDir("bin/")
		p.files("*.c")
		p.when('configurations:Release', (p) => {
			p.files("src/release/**.cpp");
		});
	});
});

File: test/premake5.lua
workspace "MyWorkspace"
	configurations { "Debug", "Release" }
	project "MyProject"
		kind "ConsoleApp"
		language "C++"
		characterset "MBCS"
		cppdialect "C++23"
		targetdir "bin/"
		files { "*.c" }
		filter "configurations:Release"
			files { "src/release/**.cpp" }
		filter {}